plugins/cpp/service/src/cppservice.cpp

```diff
@@ -88,24 +91,32 @@ namespace
       ret.range.range.endpos.column = astNode_.location.range.end.column;
 
       if (astNode_.location.file)
-      {
         ret.range.file = std::to_string(astNode_.location.file.object_id());
 
-        ret.__set_srcText(cc::util::textRange(
-          astNode_.location.file.load()->content.load()->content,
-          astNode_.location.range.start.line,
-          astNode_.location.range.start.column,
-          astNode_.location.range.end.line,
-          astNode_.location.range.end.column));
-      }
-
       TagMap::const_iterator it = _tags.find(astNode_.id);
       if (it != _tags.end())
         ret.__set_tags(it->second);
 
       return ret;
     }
 
+    /**
+     * Returns the source code text for this AST node.
+     */
+    std::string getSourceText(
```

Comment:
I think we don't need this function here. Its implementation could go to the place of its usage. (P.s.: Watch the indentation ;)